pkg/xray/provider_test.go

```diff
@@ -42,16 +44,82 @@ func getTestResty(t *testing.T) *resty.Client {
 }
 
 func testAccPreCheck(t *testing.T) {
-	restyClient := getTestResty(t)
-	// TODO check the payload and make sure it's the right license type
-	_, err := restyClient.R().Get("/artifactory/api/system/licenses/")
-	if err != nil {
-		t.Fatal(err)
-	}
 	ctx := context.Background()
-	provider, _ := testAccProviders["xray"]()
+	provider, _ := testAccProviders()["xray"]()
 	oldErr := provider.Configure(ctx, terraform.NewResourceConfigRaw(nil))
 	if oldErr != nil {
 		t.Fatal(oldErr)
 	}
 }
+
+// Create a local repository with Xray indexing enabled. It will be used in the tests
+func testAccCreateRepos(t *testing.T, repos []string) {
+	restyClient := getTestResty(t)
+
+	type Repository struct {
+		Rclass    string `json:"rclass"`
+		XrayIndex bool   `json:"xrayIndex"`
+	}
+
+	repository := Repository{}
+	repository.Rclass = "local"
+	repository.XrayIndex = true
+	for _, repo := range repos {
+		response, errRepo := restyClient.R().SetBody(repository).Put("artifactory/api/repositories/" + repo)
+		repoExists := strings.Contains(fmt.Sprint(errRepo), "Case insensitive repository key already exists")
+		if !repoExists && response.StatusCode() != http.StatusOK {
+			t.Fatal(errRepo)
+		}
+	}
+}
+
+func testAccDeleteRepos(t *testing.T, repos []string) (*resty.Response, error) {
+	restyClient := getTestResty(t)
+	for _, repo := range repos {
+		response, errRepo := restyClient.R().Delete("artifactory/api/repositories/" + repo)
```

Comment:
I am at a loss here - you have a function that deleted all repos (serially, I might add, which will be damn slow). If this is part of resource cleanup during test, then you delete the resource you have during your test.

Also, your function is hard coded to return `nil,nil` which means it might as well return nothing.

Reply:
This function either doesn't serve a clear purpose, or it's misdesigned. It seems like a bazooka and, for the record, because it's such overkill, likely to delete way more than it should

Reply:
It removes only the list of repos, submitted by the caller of that function, not everything

Reply:
mmm. then make it be like this `testAccDeleteRepos(t *testing.T, repos ... string)` so you can do
```go
testAccDeleteRepos(t,"foo","baz")
```
That will reduce syntax

Reply:
But, the `fatal` thing doesn't make sense. Can you explain it?

Reply:
No, it's basically, the same as error, according to the description. 
I've seen fatal was used [here](https://github.com/hashicorp/terraform-provider-hashicups/blob/1adc56591a26bf384f08c35ff5e3d88aea30b206/hashicups/provider_test.go#L32) for the provider test.

Reply:
Also, it's in the Artifactory provider, in `provider_test.go`

Reply:
I've implemented creating and deleting repos one by one, since we only create two of them